apps/backend/src/modules/contributors/contributor.repository.ts

```diff
@@ -136,50 +94,41 @@ class ContributorRepository implements Repository {
 			query.whereILike("contributors.name", `%${contributorName}%`);
 		}
 
-		const contributorsWithProjectsAndEmails = await query.execute();
-
-		return {
-			items: contributorsWithProjectsAndEmails.map((contributor) => {
-				return ContributorEntity.initialize(contributor);
-			}),
-		};
-	}
+		if (projectId) {
+			query.havingRaw("?? = ANY(ARRAY_AGG(projects.id))", projectId);
+		}
 
-	public async findAllWithoutPagination({
-		contributorName,
-		hasHidden = true,
-	}: {
-		contributorName?: string;
-		hasHidden?: boolean;
-	}): Promise<{ items: ContributorEntity[] }> {
-		const query = this.contributorModel
-			.query()
-			.select("contributors.*")
-			.select(
-				raw(
-					"COALESCE(ARRAY_AGG(DISTINCT jsonb_build_object('id', projects.id, 'name', projects.name)) FILTER (WHERE projects.id IS NOT NULL), '{}') AS projects",
-				),
-			)
-			.leftJoin("git_emails", "contributors.id", "git_emails.contributor_id")
-			.leftJoin("activity_logs", "git_emails.id", "activity_logs.git_email_id")
-			.leftJoin("projects", "activity_logs.project_id", "projects.id")
-			.groupBy("contributors.id")
-			.withGraphFetched("gitEmails");
+		switch (orderBy) {
+			case ContributorOrderBy.CREATED_AT: {
+				query.orderBy("created_at", SortType.DESCENDING);
+				break;
+			}
 
-		if (!hasHidden) {
-			query.whereNull("contributors.hiddenAt");
+			case ContributorOrderBy.LAST_ACTIVITY_DATE: {
+				query.orderBy("last_activity_date", SortType.DESCENDING);
+				break;
+			}
 		}
```

Comment:
Not sure about this switch. In theory I could just write something like `query.orderBy(orderBy, SortType.DESCENDING);`
But if user pass `orderBy` param with value which is not belongs to ContributorOrderBy enum, then he'll get error with DB query details, which I think is bad